wallet/backend.go

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package wallet
+
+import (
+	"fmt"
+	"io"
+
+	ed "github.com/oasisprotocol/curve25519-voi/primitives/ed25519"
+
+	"perun.network/go-perun/wallet"
+)
+
+type Backend struct{}
+
+var _ wallet.Backend = Backend{}
+
+func init() {
+	wallet.SetBackend(Backend{})
+}
+
+func (Backend) NewAddress() wallet.Address {
+	a := make(Address, 0)
```

Comment:
Why create with wrong size? The result is an invalid Address object, right? If you create with correct size, you don't need to allocate again during `UnmarshalBinary`.

Reply:
0-size allocs should be no-ops and I just want to guarantee that I have an instance of the correct type. Keeping the re-allocation in Unmarshal is just more robust overall.